tools/congestion-model/src/model/mod.rs

```diff
@@ -35,26 +38,44 @@ pub struct Model {
     // Workload state
     pub(crate) transactions: TransactionRegistry,
     pub(crate) producer: Box<dyn Producer>,
+
+    stats_writer: StatsWriter,
+    start_time: DateTime<Utc>,
 }
 
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
-pub struct ShardId(usize);
+pub struct ShardId(pub usize);
 
 impl Model {
     pub fn new(
         mut shards: Vec<Box<dyn CongestionStrategy>>,
         mut producer: Box<dyn Producer>,
+        mut stats_writer: StatsWriter,
```

Comment:
The model constructor should not need an input for the writer.
It should not care about anything other than the strategy and the workload as inputs.